framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts

```diff
@@ -121,4 +126,19 @@ export class ReportMisbehaviorPlugin extends BasePlugin {
 	private _registerAfterMiddlewares(_options: Options): void {
 		this._app.use(middlewares.errorMiddleware());
 	}
+
+	private _subscribeToChannel(): void {
+		this._channel.subscribe('app:network:event', async (info: EventInfoObject) => {
+			const {
+				data: { event, data },
+			} = info as {
+				data: { event: string; data: { block: string } };
+			};
+
+			if (event === 'postBlock') {
+				const { header } = this.codec.decodeBlock(data.block);
```

Comment:
I think here it would be nicer to decode in a js object format and use consistently below. js object will be required for the contradiction checking too